plugin.video.external.library/libs/mem_storage.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2023, Roman Miroshnychenko aka Roman V.M.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""In-memory storage"""
+
+import json
+
+import xbmcgui
+
+
+class MemStorage:
+    """
+    Stores a JSON-serializable Python object in a shared memory within Kodi
+
+    It can be used to exchange data between different Python scripts running inside Kodi.
+    """
+    def __init__(self, window_id=10000):
```

Comment:
I like the solution, but basically each and every other add-on can access it right? And perhaps overwrite values? 

I would it not make sense to prepend the add-on ID to the keys?

Reply:
Yes, the danger storage keys collision exists. I used a rather long key that was supposed to be unique but your suggestion makes sense. I included the addon ID in the key.